File: solutions/zig/darts/1/darts.zig
pub const Coordinate = struct {
    x: f32,
    y: f32,

    pub fn init(x_coord: f32, y_coord: f32) Coordinate {
        return Coordinate{ .x = x_coord, .y = y_coord };
    }

    pub fn score(self: Coordinate) usize {
        const dist = self.x * self.x + self.y * self.y;
        if (dist <= 1.0) {
            return 10;
        } else if (dist <= 25.0) {
            return 5;
        } else if (dist <= 100.0) {
            return 1;
        }

        return 0;
    }
};

File: solutions/go/darts/1/darts.go
package darts

func Score(x, y float64) int {
	var dist = x*x + y*y

	switch {
	case dist <= 1:
		return 10
	case dist <= 25:
		return 5
	case dist <= 100:
		return 1
	default:
		return 0
	}
}
